### PR TITLE
docs: move application recipes to how-to guides

### LINKED ISSUE #6106: docs: move Application recipes to how-to guides
At the moment, the [Application](https://loopback.io/doc/en/lb4/Application.html) page in "Behind the scenes" is mixing [explanations](https://documentation.divio.com/explanation/) with [how-to guides](https://documentation.divio.com/how-to-guides/). This makes it difficult for our users to find the relevant content (a recipe to achieve the outcome they want).

As part of our adoption of the four-quadrant documentation system (see #5549 and #5718), we should extract all guides from explanation pages into guide pages (see #5768 for inspiration) and keep explanation pages focused on explaining the concepts & providing background information.

This story is a part of Epic #5783 move recipes to how-to guides. 

## Acceptance criteria

Move the following recipes into "How-to guides", decide yourself whether to create a new page for each recipe or group multiple recipes in one page.

- [ ] [How to make your own application class](https://loopback.io/doc/en/lb4/Application.html#making-your-own-application-class)
- [ ] [How to configure app-level bindings](https://loopback.io/doc/en/lb4/Application.html#binding-configuration)
- [ ] How to register artifacts like [components](https://loopback.io/doc/en/lb4/Application.html#components), [controllers](https://loopback.io/doc/en/lb4/Application.html#controllers) and [servers](https://loopback.io/doc/en/lb4/Application.html#servers) to the application. 

  _I think we should have a single page for all of them, we should mention non-core artifacts like DataSources and Repositories too, and include instructions for adding artifacts via project-layout conventions, where the artifacts are picked up and registered by `@loopback/boot`._
- [ ] [How to serve static files](https://loopback.io/doc/en/lb4/Application.html#serve-static-files)

Consider adding a "Common tasks" section to "Application" page, it should list pointers to new how-to guides. See [Server >> Common tasks](https://loopback.io/doc/en/lb4/Server.html#common-tasks) for inspiration.

### PR DESCRIPTION
**Pull request by mrmodise**
Closes #6106

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/sidebars/lb4_sidebar.yml
Comment (raymondfeng) [inline docs/site/sidebars/lb4_sidebar.yml:366]: Maybe `Configuring Applications` or `Customizing Applications` is a better title to be consistent with others that use `...ing`

Comment (mrmodise) [reply docs/site/sidebars/lb4_sidebar.yml:366]: Noted. Thanks

File: docs/site/Application-setup-tips.md
Comment (dhmlau) [inline docs/site/Application-setup-tips.md:3]: I believe we're trying to use the title style where the first character of each word being upper case (though there're some inconsistencies in the docs).

```suggestion
title: 'Tips for Application Setup'
```

File: docs/site/Application-setup-tips.md
Comment (dhmlau) [inline docs/site/Application-setup-tips.md:46]: Wrapping it in warning.html, so my suggested change is below. Please preview it locally to make sure it gets rendered properly. Thanks.

```
{% include warning.html content="The static assets are served before LoopBack sequence of actions. If an error is thrown, the `reject` action will NOT be triggered." %}
```

File: docs/site/Automatic-binding.md
Comment (dhmlau) [inline docs/site/Automatic-binding.md:3]: Similar comment about the page title
```suggestion
title: 'Automatic Application Configuration'
```

File: docs/site/Configuring-applications.md
Comment (dhmlau) [inline docs/site/Configuring-applications.md:11]: duplicate content as the automatic-binding.md?

File: docs/site/Manual-binding.md
Comment (dhmlau) [inline docs/site/Manual-binding.md:3]:
```suggestion
title: 'Manually Configuring your Application'
```

**Review by dhmlau — COMMENTED** [review]
@mrmodise, thanks for the PR. Since this PR is in draft mode, I'm not sure if it's ready for review. I've put some comments on formatting-related items. Thanks.

**Comment by mrmodise** [comment]
Thanks @dhmlau for the comments. It will be ready this week. I faced issues pulling the latest master that caused some delays on completing this

File: docs/site/Application-setup-tips.md
Comment (dhmlau) [inline docs/site/Application-setup-tips.md:48]: Actually we can remove this line, because we have the "warning note". It will be highlighted.

**Review by dhmlau — APPROVED** [review]
LGTM. I've reviewed mostly from the readers' point of view, and just had a minor comment. Please make sure you get other technical reviews before landing. Thanks @mrmodise!

**Comment by mrmodise** [comment]
@dhmlau agreed, I will wait for both @bajtos and @raymondfeng to review. Particularly, around the use of the Booter class. Amazing how there is so much hidden information in LoopBack website 😄. I have purposed to surface it

**Comment by mrmodise** [comment]
@bajtos please review. Thanks

**Review by raymondfeng — APPROVED** [review]

**Pull request merged**
